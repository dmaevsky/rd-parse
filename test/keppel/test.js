const { readFileSync } = require('fs');
const { join } = require('path');
const Parser = require('../../src/index');
const grammar = require('./grammar');

describe('The keppel grammar', () => {
  it('parses a test file', () => {
    const text = readFileSync(join(__dirname, 'corpus/1.kppl'), 'utf8');

    expect(Parser.parse(grammar, text)).toMatchSnapshot();
  });

  it('parses another test file', () => {
    const text = readFileSync(join(__dirname, 'corpus/2.kppl'), 'utf8');

    expect(Parser.parse(grammar, text)).toMatchSnapshot();
  });
});
