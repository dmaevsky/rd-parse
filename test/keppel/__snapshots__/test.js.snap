// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The keppel grammar parses a test file 1`] = `
[
  {
    "type": "tag",
    "name": "body",
    "body": [
      {
        "type": "tag",
        "name": "div",
        "id": "navbar",
        "classes": [
          "navbar",
          "navbar-inverse",
          "navbar-fixed-top"
        ],
        "body": [
          {
            "type": "tag",
            "name": "div",
            "classes": [
              "navbar-inner"
            ],
            "body": [
              {
                "type": "tag",
                "name": "a",
                "attributes": [
                  {
                    "name": "href",
                    "value": "#"
                  },
                  {
                    "name": "test",
                    "value": "My test"
                  }
                ],
                "classes": [
                  "brand"
                ],
                "body": [
                  {
                    "type": "free text",
                    "value": "Awesome"
                  }
                ]
              },
              {
                "type": "tag",
                "name": "ul",
                "classes": [
                  "nav"
                ],
                "body": [
                  {
                    "type": "tag",
                    "name": "li",
                    "body": [
                      {
                        "type": "tag",
                        "name": "a",
                        "attributes": [
                          {
                            "name": "href",
                            "value": "#"
                          }
                        ],
                        "body": [
                          {
                            "type": "free text",
                            "value": "Item 1"
                          }
                        ]
                      }
                    ]
                  },
                  {
                    "type": "tag",
                    "name": "li",
                    "body": [
                      {
                        "type": "tag",
                        "name": "a",
                        "attributes": [
                          {
                            "name": "href",
                            "value": "#"
                          }
                        ],
                        "body": [
                          {
                            "type": "free text",
                            "value": "Item 2"
                          }
                        ]
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
]
`;

exports[`The keppel grammar parses another test file 1`] = `
[
  {
    "type": "free text",
    "value": "<!DOCTYPE html>"
  },
  {
    "type": "tag",
    "name": "html",
    "attributes": [
      {
        "name": "lang",
        "value": "en"
      }
    ],
    "body": [
      {
        "type": "tag",
        "name": "head",
        "body": [
          {
            "type": "tag",
            "name": "meta",
            "attributes": [
              {
                "name": "charset",
                "value": "utf-8"
              }
            ]
          },
          {
            "type": "tag",
            "name": "title",
            "body": [
              {
                "type": "free text",
                "value": "Awesome nav bar example"
              }
            ]
          },
          {
            "type": "tag",
            "name": "link",
            "attributes": [
              {
                "name": "rel",
                "value": "stylesheet"
              },
              {
                "name": "href",
                "value": "http://netdna.bootstrapcdn.com/twitter-bootstrap/2.3.1/css/bootstrap-combined.no-icons.min.css"
              }
            ]
          },
          {
            "type": "tag",
            "name": "link",
            "attributes": [
              {
                "name": "rel",
                "value": "stylesheet"
              },
              {
                "name": "href",
                "value": "http://netdna.bootstrapcdn.com/font-awesome/3.1.1/css/font-awesome.min.css"
              }
            ]
          }
        ]
      },
      {
        "type": "tag",
        "name": "body",
        "body": [
          {
            "type": "tag",
            "name": "div",
            "id": "navbar",
            "classes": [
              "navbar",
              "navbar-inverse",
              "navbar-fixed-top"
            ],
            "body": [
              {
                "type": "tag",
                "name": "div",
                "classes": [
                  "navbar-inner"
                ],
                "body": [
                  {
                    "type": "tag",
                    "name": "a",
                    "attributes": [
                      {
                        "name": "href",
                        "value": "#"
                      }
                    ],
                    "classes": [
                      "brand"
                    ],
                    "body": [
                      {
                        "type": "free text",
                        "value": "Awesome"
                      }
                    ]
                  },
                  {
                    "type": "tag",
                    "name": "ul",
                    "classes": [
                      "nav"
                    ],
                    "body": [
                      {
                        "type": "tag",
                        "name": "li",
                        "body": [
                          {
                            "type": "tag",
                            "name": "a",
                            "attributes": [
                              {
                                "name": "href",
                                "value": "#"
                              }
                            ],
                            "body": [
                              {
                                "type": "free text",
                                "value": "Item 1"
                              }
                            ]
                          }
                        ]
                      },
                      {
                        "type": "tag",
                        "name": "li",
                        "body": [
                          {
                            "type": "tag",
                            "name": "a",
                            "attributes": [
                              {
                                "name": "href",
                                "value": "#"
                              }
                            ],
                            "body": [
                              {
                                "type": "free text",
                                "value": "Item 2"
                              }
                            ]
                          }
                        ]
                      }
                    ]
                  },
                  {
                    "type": "tag",
                    "name": "ul",
                    "classes": [
                      "nav",
                      "pull-right"
                    ],
                    "body": [
                      {
                        "type": "tag",
                        "name": "li",
                        "classes": [
                          "dropdown"
                        ],
                        "body": [
                          {
                            "type": "tag",
                            "name": "a",
                            "attributes": [
                              {
                                "name": "href",
                                "value": "#"
                              }
                            ],
                            "classes": [
                              "dropdown-toggle"
                            ],
                            "body": [
                              {
                                "type": "tag",
                                "name": "i",
                                "classes": [
                                  "icon-user",
                                  "icon-white"
                                ],
                                "body": [
                                  {
                                    "type": "free text",
                                    "value": " Logged in as: Mr. Right "
                                  }
                                ]
                              },
                              {
                                "type": "tag",
                                "name": "b",
                                "classes": [
                                  "caret"
                                ]
                              }
                            ]
                          },
                          {
                            "type": "tag",
                            "name": "ul",
                            "classes": [
                              "dropdown-menu"
                            ],
                            "body": [
                              {
                                "type": "tag",
                                "name": "li",
                                "body": [
                                  {
                                    "type": "tag",
                                    "name": "a",
                                    "attributes": [
                                      {
                                        "name": "href",
                                        "value": "#"
                                      }
                                    ],
                                    "body": [
                                      {
                                        "type": "free text",
                                        "value": "Profile"
                                      }
                                    ]
                                  }
                                ]
                              },
                              {
                                "type": "tag",
                                "name": "li",
                                "classes": [
                                  "divider"
                                ]
                              },
                              {
                                "type": "tag",
                                "name": "li",
                                "body": [
                                  {
                                    "type": "tag",
                                    "name": "a",
                                    "attributes": [
                                      {
                                        "name": "href",
                                        "value": "#"
                                      }
                                    ],
                                    "body": [
                                      {
                                        "type": "free text",
                                        "value": "Sign out"
                                      }
                                    ]
                                  }
                                ]
                              }
                            ]
                          }
                        ]
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
]
`;
